drivers/sensor/lis2dux12/lis2dux12.c

```diff
@@ -0,0 +1,412 @@
+/* ST Microelectronics LIS2DUX12 3-axis accelerometer driver
+ *
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/lis2dux12.pdf
+ */
+
+#define DT_DRV_COMPAT st_lis2dux12
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <string.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/dt-bindings/sensor/lis2dux12.h>
+
+#include "lis2dux12.h"
+
+LOG_MODULE_REGISTER(LIS2DUX12, CONFIG_SENSOR_LOG_LEVEL);
+
+static int lis2dux12_set_odr(const struct device *dev, uint8_t odr)
+{
+	const struct lis2dux12_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	lis2dux12_md_t mode = {.odr = odr};
+
+	return lis2dux12_mode_set(ctx, &mode);
+}
+
+static int lis2dux12_set_range(const struct device *dev, uint8_t range)
+{
+	int err = 0;
+	struct lis2dux12_data *data = dev->data;
+	const struct lis2dux12_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+	lis2dux12_md_t val = { .odr = cfg->odr, .fs = range };
+
+	err = lis2dux12_mode_set(ctx, &val);
+
+	if (!err) {
```

Comment:
invert the logic here:

```
if (err) {
        return err;
}
```

and just `return 0;` at the end of the function, keep the normal execution flow inline, easier to follow the code intended path